### DISCUSSION TITLE
[DONE] Gradient Colormaps

### DISCUSSION THREAD

**Discussion by aliasmrchips** [developmentseed/titiler#299] [Feature Request]
How would you pass in a gradient colormap, like "viridis" (if it was not a named colormap), for example?

**Comment by vincentsarago** [comment]
@aliasmrchips for now this is not supported natively via query_parameters. In order to support this we would have to add `matplotlib` dependencies (see: https://cogeotiff.github.io/rio-tiler/colormap/#custom-colormaps) 

```python
ndvi = matplotlib.colors.LinearSegmentedColormap.from_list(
    'ndvi', [
        '#422112',
        '#724C01',
        '#CEA712',
        '#FFA904',
        '#FDFE00',
        '#E6EC06',
        '#BACF00',
        '#8BB001',
        '#72A002',
        '#5B8D03',
        '#448102',
        '#2C7001',
        '#176100',
    ],
    256,
)

x = numpy.linspace(0, 1, 256)
cmap_vals = ndvi(x)[:, :]
cmap_uint8 = (cmap_vals * 255).astype('uint8')
ndvi_dict = {idx: value.tolist() for idx, value in enumerate(cmap_uint8)}
```

previous discussion: https://github.com/developmentseed/titiler/issues/153



We could think about a user wanted to do something like:

```python
cmap = urlencode(
    {
        "colormap": json.dumps(
            {
                "1": "#FFFFFFFF",
                "10": "#AEFBFFFF",
                ...
                "255": "#ddcb9aFF",
            }
        )
    }
)
response = app.get(
    f"/cog/tiles/8/53/50.png?url=https://myurl.com/cog.tif&bidx=1&{cmap}&colormap_type=linear"
)
```

For now I can add an example https://github.com/developmentseed/titiler/tree/master/docs/examples/code

**Reply by aliasmrchips** [reply]
Thanks for the prompt response and detailed explanation! That is exactly what I would like to do... so an example would be great!

**Reply by vincentsarago** [reply]
https://github.com/developmentseed/titiler/pull/301

**Comment by aliasmrchips** [comment]
@vincentsarago Is this still the correct way to do this? I noticed that the current version gets the ColorMapType enum from rio-tiler, e.g., `from rio_tiler.types import ColorMapType`.  Thanks!

**Reply by vincentsarago** [reply]
rio-tiler's `ColorMapType` is a data type hint (https://github.com/cogeotiff/rio-tiler/blob/master/rio_tiler/types.py#L24-L27) while the `ColorMapType` showed in the example [here](https://github.com/developmentseed/titiler/blob/master/docs/examples/code/tiler_with_custom_colormap.md?plain=1#L49-L53) is an Emum which is something different. 

I don't think the example is wrong and it should still work.
